op-challenger/game/fault/trace/asterisc/provider.go

```diff
@@ -177,7 +177,7 @@ func NewTraceProviderForTest(logger log.Logger, m vm.Metricer, cfg *config.Confi
 		logger:         logger,
 		dir:            dir,
 		prestate:       cfg.AsteriscAbsolutePreState,
-		generator:      vm.NewExecutor(logger, m, cfg.Asterisc, cfg.AsteriscAbsolutePreState, localInputs),
+		generator:      vm.NewExecutor(logger, m, vm.NewOpProgramVmConfig(cfg.Asterisc), cfg.AsteriscAbsolutePreState, localInputs),
```

Comment:
Shouldn't we just call the `Config()` method on the `VmConfig` type here instead of specifying the `OpProgram` vm type?

```suggestion
		generator:      vm.NewExecutor(logger, m, cfg.Config(), cfg.AsteriscAbsolutePreState, localInputs),
```

Reply:
This is a regular `vm.Config` that we're wrapping for the sake of the test - `vm.VmConfig` is an interface that wraps `vm.Config`

Reply:
Got it I see.